DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc

```diff
@@ -249,209 +286,256 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
-              if (indexLastFilterPathModules >= sizeModulesPath) {
-                edm::LogError("Logic") << " selected index (" << indexLastFilterPathModules
-                                       << ") for last filter of path \"" << iPathName
-                                       << "\" is inconsistent with number of modules in the path (" << sizeModulesPath
-                                       << ")";
-                continue;
-              }
-              // store decision of previous filter
-              bool previousFilterAccept(true);
-              for (size_t modIdx = 0; modIdx < sizeModulesPath; ++modIdx) {
-                // each filter-bin is filled, with a 0 or 1, only when all previous filters in the path have passed
-                if (not previousFilterAccept) {
-                  break;
-                }
-                // consider only selected EDFilter modules
-                auto const& moduleLabel(hltConfigProvider_.moduleLabel(pathIndex, modIdx));
-                if (this->skipModuleByEDMType(hltConfigProvider_.moduleEDMType(moduleLabel)) or
-                    this->skipModuleByType(hltConfigProvider_.moduleType(moduleLabel))) {
-                  continue;
-                }
-                // index of the module in the path [0,sizeModulesPath)
-                const unsigned slotModule(hltConfigProvider_.moduleIndex(pathIndex, moduleLabel));
-                bool filterAccept(false);
-                if (slotModule < indexLastFilterPathModules) {
-                  filterAccept = true;
-                } else if (slotModule == indexLastFilterPathModules) {
-                  filterAccept = pathAccept;
-                }
-                LogTrace("") << "[HLTFiltersDQMonitor::analyze]         "
-                             << "HLTConfigProvider::moduleLabel(" << pathIndex << ", " << modIdx << ") = \""
-                             << moduleLabel << "\", HLTConfigProvider::moduleIndex(" << pathIndex << ", \""
-                             << moduleLabel << "\") = " << slotModule << ", filterAccept = " << filterAccept
-                             << ", previousFilterAccept = " << previousFilterAccept;
-                auto const ibin(axis->FindBin(moduleLabel.c_str()));
-                if ((0 < ibin) and (ibin <= axis->GetNbins())) {
-                  mePathTProf->Fill(ibin - 0.5, filterAccept);
-                }
-                previousFilterAccept = filterAccept;
-              }
+        auto const mePathName = efficPlotNamePrefix_ + idset + "_" + iPathName;
+
+        // consider only Paths with a MonitorElement
+        if (mePathMap_.find(mePathName) == mePathMap_.end()) {
+          LogDebug("HLTFiltersDQMonitor") << "No MonitorElement associated to Path \"" << iPathName
+                                          << "\" in Dataset \"" << idset << "\" (will be ignored)";
+          continue;
+        }
+
+        MonitorElement* const mePathProf = mePathMap_[mePathName];
```

Comment:
Also here storing the returned iterator from `mePathMap_.find(mePathName)` in a variable and de-referencing it would be more efficient.